### ISSUE TITLE
Doesn't deal well with larger subnets

### ISSUE THREAD

**Issue by passcod** [localsend/localsend#74]
It seems maybe Localsend is assuming that subnets are /24? My local subnet is a /23, and so I get situations like:
- Device A is at 10.0.0.2, gets ID `#2`
- Device B is at 10.0.1.3, gets ID `#3`
- Type `#3` into device A, it tries to connect to `10.0.0.3`, which isn't the right place
- Type `#2` into device B, it tries to connect to `10.0.1.2`, which isn't the right place

The ID should really assign from the subnet, not just take the last digit of the IPv4. So 10.0.1.3/23 should be `#258`.

Version 1.6.2 on Linux, Android, iOS.

**Comment by Tienisto** [comment]
Yes, LocalSend is currently assuming /24 because it works in most environments. Another reason is that there seems to be a bug that the subnet cannot be fetched on Windows.

Currently, the workaround is to enter the IP address manually (The button next to "Nearby Devices").

I am working on a solution to manually configure the subnet, so this should fix your problem too.

**Comment by justusthane** [comment]
It looks like the capability to get the subnet mask exists in the current beta version of Dart: https://github.com/dart-lang/sdk/issues/54237#issuecomment-5135420332

Disclaimer: I'm just an interested user; I have no familiarity with Dart.
